### PR TITLE
Adding removal of records from BCF header.

### PR DESCRIPTION
**Pull request by holtgrewe**

### CODE REVIEW

File: src/bcf/header.rs [hunk 1]
```diff
@@ -35,21 +35,35 @@ pub struct Header {
...
     }
 
+    /// Create a new `Header` using the given `HeaderView as the template.
+    ///
+    /// After construction, you can modify the header independently from the template `header`.
+    ///
+    /// # Arguments
+    ///
+    /// - `header` - The `HeaderView` to use as the template.
     pub fn with_template(header: &HeaderView) -> Self {
```
Comment (johanneskoester) [inline src/bcf/header.rs]: THis is a good opportunity to rename the function to `from_template`, following the Rust API guidelines for constructors.

File: src/bcf/header.rs [hunk 2]
```diff
@@ -35,21 +35,35 @@ pub struct Header {
...
     }
 
+    /// Create a new `Header` using the given `HeaderView` as as template, but subsetting to the
+    /// given `samples`.
+    ///
+    /// # Arguments
+    ///
+    /// - `header` - The `HeaderView` to use for the template.
+    /// - `samples` - A slice of byte-encoded (`[u8]`) sample names.
     pub fn subset_template(header: &HeaderView, samples: &[&[u8]]) -> Result<Self, SubsetError> {
```
Comment (johanneskoester) [inline src/bcf/header.rs]: This should be `from_template_subset`

Comment (holtgrewe) [reply src/bcf/header.rs]: Changed  and rebased.

File: src/bcf/header.rs (hunk 1 above)
Comment (holtgrewe) [reply src/bcf/header.rs]: done

**Pull request merged**